src/riak_kv_pb_bucket.erl

```diff
@@ -96,28 +94,60 @@ encode(Message) ->
     {ok, riak_pb_codec:encode(Message)}.
 
 %% @doc process/2 callback. Handles an incoming request message.
-process(#rpblistbucketsreq{type = Type, timeout=T, stream=S}=Req, State) ->
-    case check_bucket_type(Type) of
-        {ok, GoodType} ->
-            do_list_buckets(GoodType, T, S, Req, State);
-        error ->
-            {error, {format, "No bucket-type named '~s'", [Type]}, State}
+process(#rpblistbucketsreq{type = Type, timeout = T, stream = S} = Req, State) ->
+    Class = case S of
+        true ->
+            {riak_kv, stream_list_buckets};
+        _ ->
+            {riak_kv, list_buckets}
+    end,
+    Accept = riak_core_util:job_class_enabled(Class),
```

Comment:
If nothing else, we should have a common function _in PB_ that does this - would remove tons of repetitive code in these modules, and would mean, if the API _does_ change in 2.3, we would only need to change it in one place.